src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { Env } from 'src/env/index';
+import { JwtPayload } from './jwt-interface';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
+  constructor(
+    private readonly redisService: RedisService,
+  ) {
+    super({
+      secretOrKey: Env.JWT.AUTH_TOKEN,
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+    });
+  }
+
+  async validate(payload: JwtPayload) {
+    const { userId } = payload;
+    const client = this.redisService.getClient();
+    const jwtToken = await client.get(`jwt:${userId}`);
+    if (!jwtToken) {
+      throw new UnauthorizedException();
```

Comment:
PullProofAI

There is a potential issue with error handling in the `validate` method. If an error occurs while retrieving the JWT token from the Redis service, it is not being properly handled. Consider adding a try-catch block to handle any potential errors and provide a meaningful response to the user.
```typescript
try {
    const jwtToken = await client.get(`jwt:${userId}`);
    if (!jwtToken) {
        throw new UnauthorizedException();
    }
} catch (error) {
    // Handle the error appropriately, e.g., log the error or return a specific error message
    throw new UnauthorizedException('Failed to retrieve JWT token');
}
```


<!-- This is an auto-generated comment by PullProofAI -->